src/lib/requestManager.ts

```diff
@@ -1,9 +1,56 @@
-import axios, { Method as HTTPMethod, ResponseType, AxiosRequestConfig, AxiosResponse } from 'axios';
+import axios, { Method as HTTPMethod, ResponseType, AxiosRequestConfig, AxiosResponse, AxiosError } from 'axios';
+
+import AuthAdapter from 'adapters/authAdapter';
+import { setItem, getItem, clearItem } from 'helpers/localStorage';
+
+import { LOGIN_URL } from '../constants';
 
 export const defaultOptions: { responseType: ResponseType } = {
   responseType: 'json',
 };
 
+export function createRequestSuccessInterceptor() {
+  return function (config: AxiosRequestConfig) {
+    const userProfile = getItem('UserProfile');
+
+    if (userProfile?.auth?.access_token) {
+      config.headers.Authorization = `Bearer ${userProfile.auth.access_token}`;
+    }
+    return config;
+  };
+}
+
+export function createResponseErrorInterceptor() {
+  return async function (error: AxiosError) {
+    if (error.response?.status === 401) {
+      const userProfile = getItem('UserProfile');
+
+      clearItem('UserProfile');
+
+      if (userProfile?.auth?.refresh_token) {
+        try {
+          const response = await AuthAdapter.loginWithRefreshToken(userProfile.auth.refresh_token);
+
+          const { attributes: authInfo } = await response.data;
+
+          /* eslint-disable camelcase */
+          setItem('UserProfile', { ...userProfile, auth: authInfo });
+          /* eslint-enable camelcase */
```

Comment:
Can we remove rule disabling?
```suggestion
          setItem('UserProfile', { ...userProfile, auth: authInfo });
```